jenkins/databricks/test.sh

```diff
@@ -36,11 +61,34 @@ sudo "$(which pip)" install pytest sre_yield requests pandas pyarrow findspark p
 export SPARK_HOME=/databricks/spark
 # change to not point at databricks confs so we don't conflict with their settings
 export SPARK_CONF_DIR=$PWD
-export PYTHONPATH=$SPARK_HOME/python:$SPARK_HOME/python/pyspark/:$SPARK_HOME/python/lib/py4j-0.10.9-src.zip
-if [[ $BASE_SPARK_VERSION == "3.2.1" ]]
+# Get Python version from environment after lowercase and remove minor.
+# Something like python3.8 for DB10.4 and python3.9 for DB11.3
+sw_versions[PYTHON]=$(python --version | sed 's/[A-Z]/\L&/g;s/ //g' | cut -d . -f 1,2)
```

Comment:
we can construct this value explicitly, also db seems to use different env variables to point to different Python uses  

```bash
${PYSPARK_PYTHON} -c 'import sys; print("python{}.{}".format(sys.version_info.major, sys.version_info.minor))'
```